test/unit/deploy.js

```diff
@@ -157,36 +157,42 @@ exports['Tessel.prototype.deployScript'] = {
 
     this.tarBundle.restore();
 
-    createTemporaryDeployCode().then(function() {
+    createTemporaryDeployCode().then(() => {
       deploy.tarBundle({
-        target: deployFolder
-      }).then((bundle) => {
-        /*
-          $ t2 run app.js
-          INFO Looking for your Tessel...
-          INFO Connected to arnold over LAN
-          INFO Writing app.js to RAM on arnold (2.048 kB)...
-          INFO Deployed.
-          INFO Running app.js...
-          testing deploy
-          INFO Stopping script...
-      */
-        test.equal(bundle.length, 2048);
-        deleteTemporaryDeployCode().then(function() {
+          target: deployFolder
+        }).then(bundle => {
```

Comment:
Instead of nesting here, you should probably just `return deploy.tarBundle(...)` and then run `then` on the main promise.